docs/rep/node-agent/README.md

```diff
@@ -0,0 +1,99 @@
+# RIK - Node Agent Architecture (Riklet)
+
+This document provides a technical reflection about our component, the **node agent**.
+
+Our component will be installed on each node of a cluster and it will be composed of 4 sub-components :
+
+- **Node Manager**
+- **Node Runtime Memory**
+- **Node Proxy**
+- **Node Monitor**
+
+We will describe these components in more details throughout this document. Here is a diagram to illustrate how these components works together :
+
+![High-level view of Rikelet](assets/node-schema.png)
+
+## 1) Node Manager
+
+The Node Manager is the main sub-component of the Riklet. It should be able to :
+
+- Interprete Scheduler instructions
```

Comment:
The node manager will be responsible for receiving commands from the scheduler? Is that a correct way to rephrase that sentence?

If so, this implies 2 major architectural points:

1. The scheduler sends command to the node, i.e. it follows a client role while the node manager is the server. This means the scheduler will have to maintain a client connection with each node on the cluster. This brings some additional sub-questions:
    1. What happens when a node dies or terminates the connection with its clients?
    2. How does the scheduler know where to connect to when it wants to send a command to a node?
    3. Can any machine become a node of your cluster?
2. As a server, the node manager must provide and expose an API for clients (The scheduler) to call and use. This is something to specify early as the scheduler implementation will depend on it.

My suggestions:

* For point 1, you want to establish a node registration process/protocol, where a node would connect to the cluster controller to let it know where it is, its id, its current resources, etc. The controller can then make a decision of whether or not the node is accepted and then let the scheduler know about a new node that just joined the cluster. The scheduler can then start talking with the new node and e.g. scheduling new workloads there.
* For point 2, I suggest that you look at [gRPC](https://grpc.io/) as a way to define that API. The scheduler would become a gRPC client to each node manager gRPC server.